sztp-agent/pkg/secureagent/daemon.go

```diff
@@ -98,11 +99,11 @@ func (a *Agent) discoverBootstrapURLs() error {
 	}
 	if a.DhcpLeaseFile != "" {
 		log.Println("[INFO] User gave us the DHCP Lease File: " + a.DhcpLeaseFile)
-		url, err := a.getBootstrapURLsViaLeaseFile()
+		urls, err := dhcp.GetBootstrapURLsViaLeaseFile(a.DhcpLeaseFile, SZTP_REDIRECT_URL)
 		if err != nil {
 			return err
 		}
-		a.SetBootstrapURL(url)
+		a.SetBootstrapURL(urls[0])
```

Comment:
The agent should be set a string of URLs right?
'BootstrapURL []string' in the agent struct

Reply:
see https://github.com/opiproject/sztp/commit/2b323ff1a170efda76b50ce6e7bed94e45945267
I did a split what comes from user vs what is internal state
from user we can get: single URL or leases file/folder or network manager
internal state should always be a list